Consider KeyboardInterrupt/SystemExit at collection time (#12191)
pytest-dev/pytest: changelog/12191.bugfix.rst

> Finally fixed the issue where keyboard interrupts and system exits were not being respected during the tests collection phase in pytest! No more frustration of tests not ending correctly — happy testing! 🐞🔨🎉

Reviewer: I like this message, but to keep in tone with the rest of our change log messages:

```suggestion
Keyboard interrupts and system exits are now probably handled during the test collection.
```
